Sources/MapboxCoreNavigation/CongestionLevel.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+import MapboxDirections
+
+public typealias CongestionRange = Range<NumericCongestionLevel>
+
+extension CongestionRange {
+    public private(set) static var low: CongestionRange = CongestionRangeLow
+    public private(set) static var moderate: CongestionRange = CongestionRangeModerate
+    public private(set) static var heavy: CongestionRange = CongestionRangeHeavy
+    public private(set) static var severe: CongestionRange = CongestionRangeSevere
+
+    public static func setCongestionRanges(low: CongestionRange, moderate: CongestionRange, heavy: CongestionRange, severe: CongestionRange) {
+        precondition(low.lowerBound >= 0, "Congestion level ranges can't include negative values.")
+        precondition(low.upperBound <= moderate.lowerBound, "Values from the moderate congestion level range can't intersect with or be lower than ones from the low congestion level range.")
+        precondition(moderate.upperBound <= heavy.lowerBound, "Values from the heavy congestion level range can't intersect with or be lower than ones from the moderate congestion level range.")
+        precondition(heavy.upperBound <= severe.lowerBound, "Values from the severe congestion level range can't intersect with or be lower than ones from the heavy congestion level range.")
+        precondition(severe.upperBound <= CongestionRangeSevere.upperBound, "Congestion level ranges can't include values greater than 100.")
```

Comment:
Will it be correct if there's a gap between subsequent congestion levels? Should we instead verify if `low.upperBound == moderate.lowerBound`? Otherwise it seems that it will treat those gaps as "unknown" congestion?

Reply:
Not sure, what would be the use case for the gaps in intervals, but the "unknown" level of congestion for a value that you haven't defined for some reason looks semantically correct.